development_scripts/deconvolution_demo.py

```diff
@@ -0,0 +1,143 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Jun  5 14:43:48 2024
+
+Script demonstrating the mass transport deconvolution class & methods
+
+email a.winiwarter@imperial.ac.uk for the data
+
+@author: awiniwar
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import ixdat
+
+from pathlib import Path
+from ixdat.techniques.deconvolution import ECMSImpulseResponse
+from ixdat import Measurement
+from ixdat.techniques.ms import MSCalResult
+from spectro_inlets_quantification import Calibration
+
+ixdat.config.plugins.activate_siq()
+
+# ------------ O2 calibration
+F_o2_mscal = MSCalResult(
+    name="O2", mol="O2", mass="M32", cal_type="ECMS calibration", F=0.11755
+)
+F_o2 = F_o2_mscal.to_siq()
+
+# ------------------ Impulse response fitting of measured data + comparison with model
+# insert your folder here:
+data_folder = Path(
+    r"C:\Users\awiniwar\OneDrive - Imperial College London\Data\photo ECMS\202405 data flurin+daniele\data_demo_script"
+)  # noqa
+data_reg_cell_pulses_o2 = Measurement.read(
+    data_folder / "O2_pulses.csv",
+    reader="ixdat",
+)
+data_reg_cell_pulses_o2.plot_measurement()
+data_reg_cell_pulses_o2.set_siq_quantifier(
+    calibration=Calibration(cal_list=[F_o2]), carrier="He"
+)
+
+# prepare a figure to co-plot the measured and modelled data
+fig2, ax2 = plt.subplots(nrows=1, ncols=1)
+ax2.set_xlabel("time / [s]")
+ax2.set_ylabel("norm. intensity")
+
+# pulse length
+pulse_length = 100
+pulse_inf = 100
```

Comment:
What is the difference (in principle) between `pulse_length` and `pulse_inf`?

Reply:
I added it as a way of dealing with noisy pulses. Should maybe be removed, but I'll leave it in here until I make this into a proper tutorial after turning everything into a `Calculator`

Reply:
added some documentation in the script